services/worker-manager/src/api.js

```diff
@@ -474,6 +474,67 @@ builder.declare({
   return res.reply(worker.serializable());
 });
 
+builder.declare({
+  method: 'post',
+  route: '/workers/:workerPoolId:/:workerGroup/:workerId',
+  name: 'updateWorker',
+  title: 'Update an existing Worker',
+  category: 'Workers',
+  stability: APIBuilder.stability.stable,
+  input: 'create-worker-request.yml',
+  output: 'worker-full.yml',
+  // note that this pattern relies on workerGroup and workerId not containing `/`
+  scopes: 'worker-manager:update-worker:<workerPoolId>/<workerGroup>/<workerId>',
+  cleanPayload: cleanCreatePayload,
+  description: [
+    'Update an existing worker.  The precise behavior of this method depends',
+    'on the provider implementing the given worker pool.  Some providers',
+    'do not support updating workers at all, and will return a 400 error.',
+  ].join('\n'),
+}, async function(req, res) {
+  // TODO: test
+  const { workerPoolId, workerGroup, workerId } = req.params;
+  const workerPool = await WorkerPool.get(this.db, workerPoolId);
+  if (!workerPool) {
+    return res.reportError('ResourceNotFound',
+      `Worker pool ${workerPoolId} does not exist`, {});
+  }
+
+  // NOTE: unlike in createWorker, we allow epxires to be in the past,
+  // as a way of "immediately expiring" a worker.
+
+  const provider = this.providers.get(workerPool.providerId);
```

Comment:
Should this use `await`?

Reply:
Not in this case -- it's just a wrapper around `Map.get`.  But, good eye :)